command/command.go

```diff
@@ -354,6 +352,35 @@ func loop(app *App, termCh chan struct{}) error {
 	}
 }
 
+func postHostMetricValuesWithRetry(app *App, postValues []*mkr.HostMetricValue) (err error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 25*time.Second)
+	defer cancel()
+
+	err = app.API.PostHostMetricValues(postValues)
+	if err == nil {
+		logger.Debugf("Posting metrics succeeded.")
+		return
+	}
+	// Do not retry when first request already took long.
+	// (Typically when first request has timeout)
+	if ctxErr := ctx.Err(); ctxErr != nil {
+		logger.Warningf("Failed to post metrics value", err.Error())
+		return
+	}
```

Comment:
I recommend to use always `err`.

Reply:
Oh, this was just my mistake.  Fixed log messages.
(And our `golib/logging` https://pkg.go.dev/github.com/mackerelio/golib@v1.0.0/logging don't have `logging.Warning` 😄